### PR TITLE
Adds script to submit jobs

### PR DESCRIPTION
**Pull request by mundruid**
- Adds `api/dalton.py` as a generic Dalton API client
- Adds `api/examples/job_submission.py` to give an example on how to use the job submission and other methods from the API. 
- Adds documentation on how to submit a job

Addresses https://github.com/secureworks/dalton/issues/157

### CODE REVIEW

File: examples/submit_job.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+"""Example of how to submit a Dalton job using Python HTTP requests."""
+import requests
+import os
+
+DALTON_URL = os.getenv("DALTON_URL", "localhost")
+
+# this file needs to include the sensor configuration in yaml format. Make sure that the files are in the proper directory.
```
Comment (whartond) [inline examples/submit_job.py:7]: You can use the API to dynamically populate this if you wanted to.

To get current Suricata or Snort sensors (https://github.com/secureworks/dalton/blob/master/app/dalton.py#L2328):

DALTON_URL/dalton/controller_api/get-current-sensors/suricata
DALTON_URL/dalton/controller_api/get-current-sensors/snort

Or to get all sensors (https://github.com/secureworks/dalton/blob/master/app/dalton.py#L2361):

DALTON_URL/dalton/controller_api/get-current-sensors-json-full

Then take the the `sensor_tech` or `tech` (for all sensors JSON) value and use that to pull the YAML (https://github.com/secureworks/dalton/blob/master/app/dalton.py#L418):

DALTON_URL/dalton/controller_api/request_engine_conf?sensor=<sensor_tech

File: examples/submit_job.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# this file needs to include any custom rules you want to test
+cust = open("custom_rules.rules", "r") 
+custom_rules = cust.read()
+
+# in this example we use suricata as sensor
+data = {
+  "sensor_tech": "suricata/6.0.4/suricata.yaml",
+  "optionProdRuleset": "prod",
+  "prod_ruleset": "/opt/dalton/rulesets/suricata/suricata.rules",
```
Comment (whartond) [inline examples/submit_job.py:19]: You can dynamically pull lists of rulesets by  sensor too (https://github.com/secureworks/dalton/blob/master/app/dalton.py#L213):

DALTON_URL/dalton/controller_api/get-prod-rulesets/suricata
DALTON_URL/dalton/controller_api/get-prod-rulesets/snort

File: examples/submit_job.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  "custom_engineconf": configuration,
+  "custom_ruleset": custom_rules,
+  "teapotJob": 1
+}
+
+# test_job.pcap can be substituted with any target pcap name. 
+pcap = open('test_job.pcap', 'rb') 
+files = {"coverage-pcap0": ("test_job.pcap", pcap)}
```
Comment (whartond) [inline examples/submit_job.py:27]: Also, there is a (configurable) max number of pcap files that can be submitted at one time.  To get that limit, call (https://github.com/secureworks/dalton/blob/master/app/dalton.py#L2367-L2374):

DALTON_URL/dalton/controller_api/get-max-pcap-files

File: examples/submit_job.py (hunk 1 above)
Comment (mundruid) [reply examples/submit_job.py]: I added three API calls in the api client:

1. `get_current_sensors` implements API call to: DALTON_URL/dalton/controller_api/get-current-sensors-json-full
2. `get_sensor_tech`
  DALTON_URL/dalton/controller_api/get-current-sensors/suricata
  DALTON_URL/dalton/controller_api/get-current-sensors/snort
  
I am using these then to create the parameters for the `submit_job` API call

File: examples/submit_job.py (hunk 3 above)
Comment (mundruid) [reply examples/submit_job.py]: Added the call `get_max_pcaps` to serve the endpoint DALTON_URL/dalton/controller_api/get-max-pcap-files

**Comment by mundruid** [comment]
@whartond thanks for the review. This helped a lot to improve this contribution. As I was reading your comments, I realized that it would be worth to create an API client and use all these calls in an example. So I added:
- `api/dalton.py` as a generic Dalton API client
- `api/examples/job_submission.py` to give an example on how to use the job submission and other methods from the API. 

I think with these two additions I have addressed all your comments.

**Comment by 0x120102181f0a040a01181c** [comment]
Closes #158

**Pull request merged**